File: e2e/cypress/integration/steps/around-the-bbc/around-the-bbc-steps.js
import {Given, Then, When} from 'cypress-cucumber-preprocessor/steps';

Given(/^I visit (.*)$/, (url) => {
    cy.visit(url);
});

Given(/I set the view to (.*)$/, (view) => {
    cy.viewport(1280, 720);
});

When(/^I scroll to the footer$/, () => {
    cy.get('footer').scrollIntoView();
});

Then(/^check the around the bbc section is present$/, () => {
    return cy.get('.around-the-bbc-swimlane').should('be.visible');
});

When(/^I check link (.*) on the around the bbc module works correctly$/, (linkNumber) => {
    cy.get('.swimlane__items').children().eq(linkNumber - 1).click({force : true});
});

File: src/test/java/com/bbc/selenium/steps/aroundthebbc/AroundTheBBCSteps.java
package com.bbc.selenium.steps.aroundthebbc;

import com.bbc.selenium.utility.ScenarioContext;
import io.cucumber.java.en.Given;
import io.cucumber.java.en.Then;
import io.cucumber.java.en.When;
import org.junit.Assert;
import org.openqa.selenium.By;
import org.openqa.selenium.Dimension;
import org.openqa.selenium.JavascriptExecutor;

public class AroundTheBBCSteps {

    ScenarioContext scenarioContext = ScenarioContext.getInstanceOfScenarioContextClass();

    @Given("^I open a (.*) browser")
    public void openFirefox(String browser) {
        scenarioContext.startBrowser(browser);
    }

    @Given("I resize the window to (\\d+) x (\\d+)$")
    public void thisIsATest(int x, int y) {
        scenarioContext.getDriver().manage().window().setSize(new Dimension(x, y));
    }

    @Given("^I navigate to the future index page$")
    public void navigateToFutureIndexPage() {
        scenarioContext.getDriver().get("https://www.bbc.com/future");
    }

    @When("^I scroll to the footer$")
    public void scrollToFooter() {
        JavascriptExecutor js = (JavascriptExecutor) scenarioContext.getDriver();
        js.executeScript("window.scrollBy(0, 10000)");
    }

    @Then("^the around the bbc section is present$")
    public void assertSectionPresent() {
        Assert.assertTrue("Around the BBC section not visible", scenarioContext.getDriver().findElement(By.className("around-the-bbc-swimlane")).isDisplayed());
    }

    @Then("^I check link (.*) on the around the bbc module works correctly$")
    public void assertLinkWorks(int link) {
        scenarioContext.getDriver().findElement(By.className("around-the-bbc-swimlane"))
                .findElements(By.className("swimlane__item")).get(link - 1).click();
    }

    @Then("^I close the browser$")
    public void teardown() {
        if (scenarioContext.getDriver() != null) {
            scenarioContext.getDriver().quit();
        }
    }
}
